utils/mountinfo.c

```diff
@@ -0,0 +1,83 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "utils/mountinfo.h"
+#include "utils/utils.h"
+
+bool find_tracing_dir(char **tracing_dir)
+{
+	FILE *fp;
+	char *line = NULL, *ptr, *end, mnt[NAME_MAX];
+	size_t len;
+	int i;
+
+	fp = fopen(PROC_MOUNTINFO, "r");
+	if (fp == NULL)
+		return false;
+
+	/* First, try to find tracefs. */
+	while (getline(&line, &len, fp) > 0) {
+		if (!(ptr = strstr(line, "tracefs")))
```

Comment:
While this would work in most cases, it might not work properly if any other mount path contains the "tracefs".  You'd better use `sscanf()` to parse the mountinfo format properly - i.e. checking 9th position specifically.